src/client/features/search/index.js

```diff
@@ -33,13 +33,15 @@ class Search extends React.Component {
       showFilters: false,
       dataSources: []
     };
-    
+
       ServerAPI.datasources()
         .then(result => {
           this.setState({
             dataSources: Object.values(result)
           });
         });
+
+    window.onpopstate = (() => window.location.reload() );
```

Comment:
This workaround may function correctly, but it might not be the proper solution.  I suspect there are places in the code that aren't using the react-router Link component.